features/situation-feedback/elastic/src/main/java/org/opennms/features/situationfeedback/elastic/ElasticFeedbackRepository.java

```diff
@@ -0,0 +1,138 @@
+import org.opennms.features.situationfeedback.api.FeedbackException;
+import org.opennms.features.situationfeedback.api.FeedbackRepository;
+import org.opennms.plugins.elasticsearch.rest.bulk.BulkRequest;
+import org.opennms.plugins.elasticsearch.rest.bulk.BulkWrapper;
+import org.opennms.plugins.elasticsearch.rest.index.IndexStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+import io.searchbox.client.JestClient;
+import io.searchbox.core.Bulk;
+import io.searchbox.core.Index;
+import io.searchbox.core.Search;
+import io.searchbox.core.SearchResult;
+import io.searchbox.core.SearchResult.Hit;
+
+public class ElasticFeedbackRepository implements FeedbackRepository {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ElasticFeedbackRepository.class);
+
+    private static final String TYPE = "situation-feedback";
+
+    private final ElasticFeedbackRepositoryInitializer initializer;
+
+    private final JestClient client;
+
+    private final int bulkRetryCount;
+
+    private IndexStrategy indexStrategy;
+
+    public ElasticFeedbackRepository(JestClient jestClient, IndexStrategy indexStrategy, int bulkRetryCount, ElasticFeedbackRepositoryInitializer initializer) {
+        this.client = jestClient;
+        this.indexStrategy = indexStrategy;
+        this.bulkRetryCount = bulkRetryCount;
+        this.initializer = initializer;
+    }
+
+    @Override
+    public void persist(Collection<AlarmFeedback> feedback) throws FeedbackException {
+        ensureInitialized();
+        if (LOG.isDebugEnabled()) {
+            for (AlarmFeedback fb : feedback) {
+                LOG.debug("Persiting {} feedback.", fb);
+            }
+        }
+
+        List<FeedbackDocument> feedbackDocuments = feedback.stream().map(FeedbackDocument::from).collect(Collectors.toList());
+        BulkRequest<FeedbackDocument> bulkRequest = new BulkRequest<>(client, feedbackDocuments, (documents) -> {
+            final Bulk.Builder bulkBuilder = new Bulk.Builder();
+            for (FeedbackDocument document : documents) {
+                final String index = indexStrategy.getIndex(TYPE, Instant.ofEpochMilli(document.getTimestamp()));
+                final Index.Builder indexBuilder = new Index.Builder(document).index(index).type(TYPE);
+                bulkBuilder.addAction(indexBuilder.build());
+            }
+            return new BulkWrapper(bulkBuilder);
+        }, bulkRetryCount);
+        // the bulk request considers retries
+        try {
+            bulkRequest.execute();
+        } catch (IOException e) {
+            LOG.error("Failed to persist feedback [{}]: {}", feedback, e.getMessage());
+            throw new FeedbackException("Failed to persist feedback", e);
+        }
+    }
+
+    @Override
+    public Collection<AlarmFeedback> getFeedback(String situationKey) throws FeedbackException {
+        Gson gson = new Gson();
```

Comment:
The gson instance is thread-safe, so we can declare it as a member variable instead of recreating it everytime here.